opm/autodiff/BlackoilModelEbos.hpp

```diff
@@ -1045,150 +1046,100 @@ namespace Opm {
-                        pres[region] += pv[c] * fs.pressure(FluidSystem::oilPhaseIdx).value();
-                    }
-                }
-                for (int c = 0; c < nc; ++c) {
-                    const int region = fipnum[c] - 1;
-                    if (region != -1) {
-
-                        fip_.fip[FIPDataType::FIP_PV][c] = pv[c];
-                        const auto& intQuants = *ebosSimulator_.model().cachedIntensiveQuantities(c, /*timeIdx=*/0);
-                        const auto& fs = intQuants.fluidState();
-                        const double hydrocarbon = fs.saturation(FluidSystem::oilPhaseIdx).value() + fs.saturation(FluidSystem::gasPhaseIdx).value();
-
-                        //Compute hydrocarbon pore volume weighted average pressure.
-                        //If we have no hydrocarbon in region, use pore volume weighted average pressure instead
-                        if (hcpv[region] != 0) {
-                            fip_.fip[FIPDataType::FIP_WEIGHTED_PRESSURE][c] = pv[c] * fs.pressure(FluidSystem::oilPhaseIdx).value() * hydrocarbon / hcpv[region];
-                        } else {
-                            fip_.fip[FIPDataType::FIP_WEIGHTED_PRESSURE][c] = pres[region] / pv[c];
-                        }
-
-                        values[region][FIPDataType::FIP_PV] += fip_.fip[FIPDataType::FIP_PV][c];
-                        values[region][FIPDataType::FIP_WEIGHTED_PRESSURE] += fip_.fip[FIPDataType::FIP_WEIGHTED_PRESSURE][c];
+                        values[region][FIPData::FIP_DISSOLVED_GAS] += fip_.fip[FIPData::FIP_DISSOLVED_GAS][c];
+                        values[region][FIPData::FIP_VAPORIZED_OIL] += fip_.fip[FIPData::FIP_VAPORIZED_OIL][c];
                     }
                 }
             }
-            else
-            {
-#if HAVE_MPI
-                // mask[c] is 1 if we need to compute something in parallel
-                const auto & pinfo =
-                        boost::any_cast<const ParallelISTLInformation&>(istlSolver().parallelInformation());
-                const auto& mask = pinfo.updateOwnerMask( fipnum );
 
-                auto comm = pinfo.communicator();
-                // Compute the global dims value and resize values accordingly.
-                dims = comm.max(dims);
-                values.resize(dims, std::vector<double>(FIPDataType::fipValues,0.0));
+            std::vector<double> hcpv(dims, 0.0);
+            std::vector<double> pres(dims, 0.0);
 
-                //Accumulate phases for each region
-                for (int phase = 0; phase < maxnp; ++phase) {
-                    for (int c = 0; c < nc; ++c) {
-                        const int region = fipnum[c] - 1;
-                        if (region != -1 && mask[c]) {
-                            values[region][phase] += fip_.fip[phase][c];
-                        }
-                    }
+            elemIt = elemCtx.gridView().template begin</*codim=*/0>();
```

Comment:
ok: I've changed this so that the element iterators are created in the first part of the `for` statements. the result is harder to read IMO, so I would not mind if you requested to revert that patch before the PR gets merged.